### ISSUE TITLE
Running roda-core from command line doesn't init RodaCoreFactory

### ISSUE THREAD

**Issue by rui-castro** [keeps/roda#547]
## Description

Running roda-core jar from command line doesn't init RodaCoreFactory properly. 
The log shows this error:
```
com.typesafe.config.ConfigException$IO: reference.conf @ jar:file:/home/tobias/work/keeps/roda.git/roda-core/roda-core/target/roda-core-2.0.0-SNAPSHOT-jar-with-dependencies.jar!/reference.conf: read error: Stream closed
```
The variable `RodaCoreFactory.ldapUtility` is `null`.

**Running RodaCoreFactory from inside IntelliJ works fine. So, maybe the problem has something to do with the way roda-core-...with-dependencies.jar is produced.**

### Steps to reproduce

1. `mvn -Proda-core-jar clean package`
2. `java -jar roda-core/roda-core/target/roda-core-2.0.0-SNAPSHOT-jar-with-dependencies.jar`

The `roda-core.log` shows:
```
2016-09-16 18:34:58,453 [main] DEBUG org.roda.core.RodaCoreFactory - RODA HOME is /home/tobias/.roda
2016-09-16 18:34:58,804 [main] INFO  org.roda.core.RodaCoreFactory - Copy files from classpath prefix=config/, destination=/home/tobias/.roda/example-config, removePrefix=true, excludePaths=[config/ldap, config/index], #resources=152
2016-09-16 18:34:58,856 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating essential directories
2016-09-16 18:34:58,921 [main] DEBUG org.roda.core.RodaCoreFactory - Finished loading roda-core.properties & roda-core-formats.properties
2016-09-16 18:34:58,933 [main] DEBUG org.roda.core.RodaCoreFactory - Finished loading roda-roles.properties
2016-09-16 18:34:58,941 [main] DEBUG org.roda.core.RodaCoreFactory - Going to instantiate Filesystem on '/home/tobias/.roda/data/storage'
2016-09-16 18:34:58,946 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating storage...
2016-09-16 18:34:58,960 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating model...
2016-09-16 18:34:58,960 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating storage & model
2016-09-16 18:34:59,176 [main] INFO  org.roda.core.RodaCoreFactory - Copy files from classpath prefix=config/index/, destination=/tmp/index6613177610829922652, removePrefix=false, excludePaths=[], #resources=121
2016-09-16 18:34:59,255 [main] INFO  org.roda.core.RodaCoreFactory - Using SOLR home: /tmp/index6613177610829922652/config/index
2016-09-16 18:35:04,884 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating solr & index
2016-09-16 18:35:04,895 [main] INFO  org.roda.core.plugins.PluginManager - PluginManager init OK
2016-09-16 18:35:04,897 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating plugin manager
2016-09-16 18:35:05,217 [main] ERROR org.roda.core.RodaCoreFactory - Error instantiating RodaCoreFactory
com.typesafe.config.ConfigException$IO: reference.conf @ jar:file:/home/tobias/work/keeps/roda.git/roda-core/roda-core/target/roda-core-2.0.0-SNAPSHOT-jar-with-dependencies.jar!/reference.conf: read error: Stream closed
	at com.typesafe.config.impl.Tokenizer$TokenIterator.nextCharRaw(Tokenizer.java:159)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.pullUnquotedText(Tokenizer.java:340)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.pullNextToken(Tokenizer.java:637)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.queueNextToken(Tokenizer.java:660)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.next(Tokenizer.java:678)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.next(Tokenizer.java:63)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.popToken(ConfigDocumentParser.java:45)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.nextToken(ConfigDocumentParser.java:51)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.checkElementSeparator(ConfigDocumentParser.java:124)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:462)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:247)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:405)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:247)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:405)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseValue(ConfigDocumentParser.java:247)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:405)
	at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:595)
	at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
	at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:307)
	at com.typesafe.config.impl.Parseable$ParseableResources.rawParseValue(Parseable.java:747)
	at com.typesafe.config.impl.Parseable$ParseableResources.rawParseValue(Parseable.java:710)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
	at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
	at com.typesafe.config.impl.Parseable.parse(Parseable.java:299)
	at com.typesafe.config.impl.ConfigImpl$1.call(ConfigImpl.java:373)
	at com.typesafe.config.impl.ConfigImpl$1.call(ConfigImpl.java:367)
	at com.typesafe.config.impl.ConfigImpl$LoaderCache.getOrElseUpdate(ConfigImpl.java:65)
	at com.typesafe.config.impl.ConfigImpl.computeCachedConfig(ConfigImpl.java:92)
	at com.typesafe.config.impl.ConfigImpl.defaultReference(ConfigImpl.java:367)
	at com.typesafe.config.ConfigFactory.defaultReference(ConfigFactory.java:413)
	at akka.actor.ActorSystem$Settings.<init>(ActorSystem.scala:160)
	at akka.actor.ActorSystemImpl.<init>(ActorSystem.scala:522)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:142)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:119)
	at akka.actor.ActorSystem$.create(ActorSystem.scala:67)
	at akka.actor.ActorSystem.create(ActorSystem.scala)
	at org.roda.core.plugins.orchestrate.AkkaEmbeddedPluginOrchestrator.<init>(AkkaEmbeddedPluginOrchestrator.java:98)
	at org.roda.core.RodaCoreFactory.instantiateMasterNodeSpecificObjects(RodaCoreFactory.java:702)
	at org.roda.core.RodaCoreFactory.instantiateNodeSpecificObjects(RodaCoreFactory.java:664)
	at org.roda.core.RodaCoreFactory.instantiate(RodaCoreFactory.java:272)
	at org.roda.core.RodaCoreFactory.instantiateMaster(RodaCoreFactory.java:207)
	at org.roda.core.RodaCoreFactory.instantiate(RodaCoreFactory.java:196)
	at org.roda.core.RodaCoreFactory.main(RodaCoreFactory.java:1447)
Caused by: java.io.IOException: Stream closed
	at java.util.zip.InflaterInputStream.ensureOpen(InflaterInputStream.java:67)
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:142)
	at java.io.FilterInputStream.read(FilterInputStream.java:133)
	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.io.BufferedReader.read(BufferedReader.java:182)
	at com.typesafe.config.impl.Tokenizer$TokenIterator.nextCharRaw(Tokenizer.java:156)
	... 46 common frames omitted
2016-09-16 18:35:05,220 [main] INFO  org.roda.core.RodaCoreFactory - RODA Core loading completed with some errors!!! See logs because these errors might cause instability in the system.
```

**Comment by hsilva-keep** [comment]
@rui-castro Can't reproduce. I've just tried it out in master branch (dba78ef0d2162fdb92993040697829509b0c2a74) & it runs flawlessly. Please confirm that you didn't introduced any change that could lead to the behaviour you're mentioning.
```bash
$> git checkout master && git branch -vvv
Already on 'master'
Your branch is up-to-date with 'origin/master'.
  hsilva_dev     555d565 [origin/hsilva_dev: adiante de 18] fixing body missing from processed email notification
  lf_dev         0231ec5 [origin/lf_dev: behind 127] adding facet filters to selected items from async table cell
* master         dba78ef [origin/master] Fixed issue related to error handling in descriptive metadata creation
  mferreira_dev  26c96ba [origin/mferreira_dev: behind 602] revised readme
  nvieira_dev    817fe41 [origin/nvieira_dev: behind 118] generic job running on a list of objects
  sleroux-dev    cb24b35 [origin/sleroux-dev: ahead 1, behind 85] misc: minor fixes after integrating sleroux code
  vfernandes_dev 4e779b0 [origin/vfernandes_dev: behind 618] wui: UserLog search

$> mvn -Proda-core-jar,core clean package -Dmaven.test.skip=true

$> java -jar roda-core/roda-core/target/roda-core-2.0.0-SNAPSHOT-jar-with-dependencies.jar index list users
index list users
	User [groups=[guests], email=demo-guest@roda-community.org, guest=false, ipAddress=, ]
	User [groups=[administrators], email=demo-admin@roda-community.org, guest=false, ipAddress=, ]

$> tail -n 10 ~/.roda/log/roda-core.log 
2016-09-19 08:11:52,244 [main] INFO  org.roda.core.RodaCoreFactory - Role ingest.transfer already exists.
2016-09-19 08:11:52,245 [main] INFO  org.roda.core.RodaCoreFactory - Role browse.transferred_resource already exists.
2016-09-19 08:11:52,245 [main] INFO  org.roda.core.RodaCoreFactory - Role browse.file already exists.
2016-09-19 08:11:52,246 [main] INFO  org.roda.core.RodaCoreFactory - Role browse.member already exists.
2016-09-19 08:11:52,246 [main] INFO  org.roda.core.RodaCoreFactory - Role browse.job already exists.
2016-09-19 08:11:52,247 [main] INFO  org.roda.core.RodaCoreFactory - Role browse already exists.
2016-09-19 08:11:52,247 [main] INFO  org.roda.core.RodaCoreFactory - Role browse.format already exists.
2016-09-19 08:11:52,248 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating node specific objects
2016-09-19 08:11:52,254 [main] DEBUG org.roda.core.RodaCoreFactory - Finished instantiating default objects
2016-09-19 08:11:52,254 [main] INFO  org.roda.core.RodaCoreFactory - RODA Core loading completed with success!
```

**Comment by luis100** [comment]
@rui-castro  Also could not reproduce.

**Comment by luis100** [comment]
@hsilva-keep do we still need shading here?

**Comment by rui-castro** [comment]
@luis100 @hsilva-keep apparently the commit that is causing this problem is https://github.com/keeps/roda/commit/b5960e701ab1d87b0c0b76e022636457dbe26d65, but I don't understand why.

**Comment by hsilva-keep** [comment]
I managed to reproduce the problem, and actually it doesn't matter in which branch/commit you are, as it always happens when RODA_HOME doesn't exist (first init).

**Comment by rui-castro** [comment]
Yesterday I've tried with master and it worked. With the HEAD of lf_dev
didn't.

Em 20/09/2016 09:33, "Hélder Silva" <notifications@github.com> escreveu:

**Comment by hsilva-keep** [comment]
@rui-castro you've tried but probably you didn't removed your RODA_HOME folder.

**Comment by hsilva-keep** [comment]
This has been solved in bb550ecdcc75c2f983d49f7213d7b0c705477dfe. The PluginManager & the usage of a library called Reflections seem to be the cause of the problem. 
But, as no simple solution was found (at least for now), I've changed the order of the init in RodaCoreFactory in order to start the PluginManager in last place & that does the trick, i.e., makes everything initialize properly.
